pkg/controller/shared_state.go

```diff
@@ -81,23 +82,35 @@ type SharedState struct {
 
 func NewSharedState(driverName, objectNamespace, topologyKey string, eventRecorder record.EventRecorder, cachedClient client.Client, azClient azdisk.Interface, kubeClient kubernetes.Interface, crdClient crdClientset.Interface) *SharedState {
 	newSharedState := &SharedState{
-		driverName:      driverName,
-		objectNamespace: objectNamespace,
-		topologyKey:     topologyKey,
-		eventRecorder:   eventRecorder,
-		cachedClient:    cachedClient,
-		crdClient:       crdClient,
-		azClient:        azClient,
-		kubeClient:      kubeClient,
-		conditionWatcher: watcher.New(context.Background(),
-			azClient, azdiskinformers.NewSharedInformerFactory(azClient, consts.DefaultInformerResync), objectNamespace),
+		driverName:             driverName,
+		objectNamespace:        objectNamespace,
+		topologyKey:            topologyKey,
+		eventRecorder:          eventRecorder,
+		cachedClient:           cachedClient,
+		crdClient:              crdClient,
+		azClient:               azClient,
+		kubeClient:             kubeClient,
+		conditionWatcher:       watcher.New(context.Background(), azClient, azdiskinformers.NewSharedInformerFactory(azClient, consts.DefaultInformerResync), objectNamespace),
 		azureDiskCSITranslator: csitranslator.NewAzureDiskCSITranslator(),
 	}
 	newSharedState.createReplicaRequestsQueue()
 
 	return newSharedState
 }
 
+func (c *SharedState) isDriverUninstall(ctx context.Context) bool {
+	w, _ := workflow.GetWorkflowFromContext(ctx)
```

Comment:
The workflow doesn't seem to be used in this function. Do we need to call `GetWorkflowFromContext` here?

Reply:
Ah, sorry I missed the call to the workflow logger below.